mantidproject/mantid
# Indirect - Fitting - ExtractQENSMembers Algorithm

Currently, the routine for extracting the fit members from QENS fitting is done within the ConvolutionFitSequential algorithm. This routine only accounts for a delta function and 1/2 Lorentzian fits and is only accessible within the ConvolutionFitSequential algorithm.

This PR moves this routine into a separate algorithm 'ExtractQENSMembers', which extracts the members from a QENS fit (once processed with the ProcessIndirectFitParameters algorithm). This algorithm will extract fit members from any fit type and can also be used with the output of other Indirect Fitting algorithms (e.g. IqtFit).

**To test:**
1. Navigate to Interfaces > Indirect > Data Analysis > ConvFit
2. Provide the supplied \*\_red file into the Sample field.
3. Provide the supplied \*\_res file into the Resolution field.
4. In the drop-down menu select 'One Lorentzian'.
5. Check the check-box labelled 'Delta Function'.
6. Ensure the check-box labelled 'Extract Members' is checked.
7. Click 'Run'.
8. Ensure a group workspace labelled '_Members' is created.
9. Ensure '\*\_Data', '\*\_Calc', '\*\_Diff',  '\*\_DeltaFunction' and '\*\_Lorentzian' workspaces are contained within the members group workspace.
10. Run the unit tests in ExtractQENSMembersTest and ensure they pass.

**Test Files:**
[ConvFit.zip](https://github.com/mantidproject/mantid/files/1678824/ConvFit.zip)



Fixes #21648. 

**Release Notes** 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
